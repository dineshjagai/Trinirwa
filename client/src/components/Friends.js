/* eslint-disable max-len */
import React, { useState, useEffect } from 'react';
import './Followers.css';
import PersonAddDisabledIcon from '@material-ui/icons/PersonAddDisabled';
import { useHistory } from 'react-router-dom';
import Friend from './Friend';
import { blockFollower, getFriends, unfollowUser } from './Module';
import ScrollDialog from './DisplayerDialog';
import { getCurrentUsername } from '../auth/authServices';

export default function Friends() {
  const history = useHistory();
  const user = getCurrentUsername();
  const [friends, setFriends] = useState([]);
  useEffect(() => {
    getFriends(user).then((result) => {
      setFriends(result.data.friends);
    });
  }, []);

  const handleB = (username) => {
    blockFollower(user, username).then((res) => {
      console.log(res.message);
      getFriends(user).then((result) => {
        setFriends(result.data.friends);
      });
    });
  };

  const handleU = (username) => {
    unfollowUser(user, username).then((res) => {
      console.log(res.message);
      getFriends(user).then((result) => {
        setFriends(result.data.friends);
      });
    });
  };

  const handleMessage = (username) => {
    console.log(username);
    localStorage.setItem('receiver', JSON.stringify(username));
    history.push('/message');
  };

  const items = friends.map((e) => <Friend info={e} handleMessage={handleMessage} handleB={handleB} handleU={handleU} />);
  return (
    <div className="box-container">
      <div className="buttonFollowers">
        <ScrollDialog
          secondary="friends"
          title="All friends"
          secondTitle="All friends"
          getFunction={getFriends}
          Icon={PersonAddDisabledIcon}
          handle={handleU}
          iconText="Unfollow"
        />
      </div>
      <div className="title">Friends</div>
      <br />
      <div className="box">
        <div className="containers">
          {items}
        </div>
      </div>
    </div>
  );
}
