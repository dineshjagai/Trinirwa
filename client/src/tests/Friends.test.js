import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Friends from '../components/Friends';

test('Test Friends snapshot', () => {
  const component = renderer.create(<Friends />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
