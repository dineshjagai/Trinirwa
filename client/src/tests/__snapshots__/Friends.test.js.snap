// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Friends snapshot 1`] = `
<div
  className="box-container"
>
  <div
    className="buttonFollowers"
  >
    <div>
      <button
        className="btn"
        id="delete"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "inherit",
            "border": "none",
            "color": "#00695c",
          }
        }
        type="button"
      >
        All friends
      </button>
    </div>
  </div>
  <div
    className="title"
    style={
      Object {
        "color": "#0C8367",
      }
    }
  >
    Friends
  </div>
  <br />
  <div
    className="box"
  >
    <div
      className="containers"
    />
  </div>
</div>
`;
